src/mame/drivers/hp2640.cpp

```diff
@@ -1012,19 +1049,7 @@ void hp2640_base_state::cpu_mem_map(address_map &map)
 {
 	map.unmap_value_low();
 	map(0x0000, 0x57ff).rom();
-	map(0x8100, 0x8100).r(m_uart, FUNC(ay51013_device::receive));
-	map(0x8120, 0x8120).r(FUNC(hp2640_base_state::async_status_r));
-	map(0x8140, 0x8140).w(FUNC(hp2640_base_state::async_control_w));
-	map(0x8160, 0x8160).w(m_uart, FUNC(ay51013_device::transmit));
-	map(0x8300, 0x8300).w(FUNC(hp2640_base_state::kb_led_w));
-	map(0x8300, 0x830d).r(FUNC(hp2640_base_state::kb_r));
-	map(0x830e, 0x830e).r(FUNC(hp2640_base_state::switches_ah_r));
-	map(0x830f, 0x830f).r(FUNC(hp2640_base_state::datacomm_sw_r));
-	map(0x8320, 0x8320).w(FUNC(hp2640_base_state::kb_prev_w));
-	map(0x8380, 0x8380).rw(FUNC(hp2640_base_state::switches_jr_r), FUNC(hp2640_base_state::kb_reset_w));
-	map(0x83a0, 0x83a0).r(FUNC(hp2640_base_state::switches_sz_r));
-	map(0x8700, 0x8700).w(FUNC(hp2640_base_state::cx_w));
-	map(0x8720, 0x8720).w(FUNC(hp2640_base_state::cy_w));
+	map(0x8000, 0x8fff).m(m_io_bank, FUNC(address_map_bank_device::amap8));
```

Comment:
Can the address map bank device be avoided by using the new “views” feature of the memory system?